### DISCUSSION TITLE
enhancement: a bufferline mode that shows only the current buffer information

### DISCUSSION THREAD

**Discussion by z0al** [helix-editor/helix#7880] [Ideas] [answered]
### Background

As of now, `editor.bufferline` accepts the following options:

- `never` (default)
- `always`
- `multiple`: only shown if more than one buffer is in use

Some of us like to see the file info on top of the buffer without introducing the clutter of "editor tabs". Ultimately, I want to emulate VS Code behavior when `showTabs=false` 👇🏻 

![image](https://github.com/helix-editor/helix/assets/12673605/568a700f-32dc-46c2-92bf-22cb48f8a35e)

But I'm happy to settle for the current Helix bufferline UX as long as _only the current buffer information is shown at all times_.

### Proposal

Introduce a new option for `editor.bufferline` e.g. `single` (or `curent-buffer-only` to be more descriptive) that when set:
- Always shows the bufferline like. Similar to `always`
- Only ever show the current buffer info even if multiple buffers are in use.

**Comment by pascalkuthe** [comment]
We already show the name of the current buffer in the statueline. The buffer line is intended to provide an overview of all open buffers, not to replace the statusline

**Reply by z0al** [reply]
> We already show the name of the current buffer in the statueline
> provide an overview of all open buffers, not to replace the statusline

That's right. The bufferline already shows the file name as you stated. But It's possible to remove the `file-name` from the statusline config already. 

Do you see any harm in giving the user more control over the behavior of the bufferline (current buffer only vs all buffers)? Neovim plugins like [bufferline.nvim](https://github.com/akinsho/bufferline.nvim) offer similar functionality with a bit of config.

I understand if this is something you want to keep out of the core and leave for Plugins later. I just thought it was a low effort today as it's about rendering less information than we already do.

**Reply by z0al** [reply, marked as answer]
Thinking about this again. I think what I'm really missing here is neovim's [winbar](https://www.youtube.com/watch?v=LKW_SUucO-k) equivalent rather than a "buffer" line.

**Comment by z0al** [comment]
Closing. I will open a new discussion for a winbar & global status line later.
